lib/graphql/anycable/cleaner.rb

```diff
@@ -48,6 +52,27 @@ def clean_events
         end
       end
 
+      def clean_fingerprint_subscriptions
+        redis.scan_each(match: "#{adapter::SUBSCRIPTIONS_PREFIX}*") do |key|
+          redis.smembers(key).each do |subscription_id|
+            next if redis.exists?(adapter::SUBSCRIPTION_PREFIX + subscription_id)
+
+            redis.srem(key, subscription_id)
+          end
+        end
+      end
+
+      def clean_topic_fingerprints
+        redis.scan_each(match: "#{adapter::FINGERPRINTS_PREFIX}*") do |key|
+          redis.zremrangebyscore(key, '-inf', '0')
+          redis.zmembers(key).each do |fingerprint|
```

Comment:
do you mean:
```suggestion
          redis.smembers(key).each do |fingerprint|
```
?

Reply:
No, I changed `graphql-fingerprints` from Set to Sorted Set to keep track of number of subscriptions for every fingerprint in its score to remove elements only when there are no more subscriptions for every given fingerprint.

Reply:
Mmmh Redis does not know this command:

`ERR unknown command `zmembers`, with args beginning with: `graphql-fingerprints::`

Reply:
Oh, it should be `zrange(key, 0, -1)`. Fixed.

However I wish that there would be `zmembers` as shorthand for it to get all members at once. :sweat_smile: